File: python-code/ex3/contaBancaria.py
class ContaBancaria:
  def __init__(self, titular, saldo_inicial=0):
    self.__titular = titular
    self.__saldo = saldo_inicial

  def depositar(self, valor):
    if valor > 0:
      self.__saldo += valor
      print(f"Depositado: R$ {valor:.2f}. Saldo atual: R$ {self.__saldo:.2f}")
    else:
      print("Valor do depósito deve ser positivo.")

  def sacar(self, valor):
    if valor > 0:
      if valor <= self.__saldo:
        self.__saldo -= valor
        print(f"Sacado: R$ {valor:.2f}. Saldo atual: R$ {self.__saldo:.2f}")
      else:
        print("Saldo insuficiente.")
    else:
      print("Valor do saque deve ser positivo.")

  def exibir_saldo(self):
    print(f"Saldo atual: R$ {self.__saldo:.2f}")

conta = ContaBancaria("Lucas", 2000)
conta.depositar(400)
conta.sacar(600)
conta.exibir_saldo()

File: java-code/src/main/java/ex3/ContaBancaria.java
package ex3;

public class ContaBancaria {
    private double saldo;
    private String titular;

    public ContaBancaria(String titular, double saldoInicial) {
        this.titular = titular;
        this.saldo = saldoInicial;
    }

    public void depositar(double valor) {
        if (valor > 0) {
            saldo += valor;
            System.out.printf("Depositado: R$ %.2f. Saldo atual: R$ %.2f%n", valor, saldo);
        } else {
            System.out.println("Valor do depósito deve ser positivo.");
        }
    }

    public void sacar(double valor) {
        if (valor > 0) {
            if (valor <= saldo) {
                saldo -= valor;
                System.out.printf("Sacado: R$ %.2f. Saldo atual: R$ %.2f%n", valor, saldo);
            } else {
                System.out.println("Saldo insuficiente.");
            }
        } else {
            System.out.println("Valor do saque deve ser positivo.");
        }
    }

    public void exibirSaldo() {
        System.out.printf("Saldo atual: R$ %.2f%n", saldo);
    }
}
